src/rdf4cpp/rdf/datatypes/owl/Rational.cpp

```diff
@@ -48,6 +59,35 @@ std::partial_ordering capabilities::Comparable<owl_rational>::compare(cpp_type c
     }
 }
 
+template<>
+nonstd::expected<capabilities::Numeric<owl_rational>::abs_result_cpp_type, DynamicError> capabilities::Numeric<owl_rational>::abs(cpp_type const &operand) noexcept {
+    return boost::multiprecision::abs(operand);
+}
+
+template<>
+nonstd::expected<capabilities::Numeric<owl_rational>::round_result_cpp_type, DynamicError> capabilities::Numeric<owl_rational>::round(cpp_type const &operand) noexcept {
+    auto const integral_part = numerator(operand) / denominator(operand);
+    cpp_type const fractional_part = operand - integral_part;
+
+    if (numerator(fractional_part) * 2 < denominator(fractional_part)) {
+        return integral_part;
+    }
+
+    return integral_part + 1;
+}
+
+template<>
+nonstd::expected<capabilities::Numeric<owl_rational>::floor_result_cpp_type, DynamicError> capabilities::Numeric<owl_rational>::floor(cpp_type const &operand) noexcept {
+    return static_cast<boost::multiprecision::cpp_int>(operand);
+}
+
+template<>
+nonstd::expected<capabilities::Numeric<owl_rational>::ceil_result_cpp_type, DynamicError> capabilities::Numeric<owl_rational>::ceil(cpp_type const &operand) noexcept {
+    auto const integral = static_cast<boost::multiprecision::cpp_int>(operand);
+    auto const rest = numerator(operand) % denominator(operand);
```

Comment:
I don't get why the rest thing is necessary. Shouldn't casting to integral simply result in an ceil already?

Reply:
Casting to integral is usually a floor not a ceil. Thats why the rest thingy is there